cobalt/watchdog/watchdog.cc

```diff
@@ -579,23 +579,46 @@ std::string Watchdog::GetWatchdogViolations(
             violations_count_ -= violations_count;
             if (!static_cast<base::DictionaryValue*>(violations_map_.get())
                      ->empty()) {
-              WriteWatchdogViolations();
+              pending_write_ = true;
```

Comment:
Why the change here?

Reply:
We don't want to write every single time there's a change. This can cause I/O limitations on certain platforms. Especially here when it could possibly write once per fetched client (100 fetched clients is an additional 100 writes). It will write when it can (given the backoff) in the monitor loop or when Watchdog / Cobalt is terminated or when it's absolutely necessary. Watchdog otherwise operates with the in-memory data structures.